drivers/pwm/pwm_mcux_sctimer.c

```diff
@@ -95,11 +144,21 @@ static int mcux_sctimer_pwm_set_cycles(const struct device *dev,
 
 		LOG_DBG("SETUP dutycycle to %u\n", duty_cycle);
 		data->channel[channel].dutyCyclePercent = duty_cycle;
-		if (SCTIMER_SetupPwm(config->base, &data->channel[channel],
-				     kSCTIMER_EdgeAlignedPwm, pwm_freq,
-				     clock_freq, &data->event_number[channel]) == kStatus_Fail) {
-			LOG_ERR("Could not set up pwm");
-			return -ENOTSUP;
+		if (data->event_number[channel] !=
```

Comment:
Hi @morandg, first- apologies for the delay in revisiting this PR, and thank you for the contribution.

I've tested this PR on an LPC55S69, using `tests/drivers/pwm/pwm_api`, and I still see the period remaining constant in both tests. I believe the issue is here: https://github.com/zephyrproject-rtos/zephyr/pull/58147/files#diff-dd625dd46514efba61daa8b5a67ffb1d559a3c90c3e9f21fcb918792cd70c4cdR125. When I change the `&&` condition to `||` locally, the output pwm period does change from 750Hz to 500Hz.